internal/manager/config_validation.go

```diff
@@ -35,9 +38,7 @@ func namespacedNameFromFlagValue(flagValue string) (OptionalNamespacedName, erro
 }
 
 func gatewayAPIControllerNameFromFlagValue(flagValue string) (string, error) {
```

Comment:
I just wanted to call out 1 thing: this `func` is not called anywhere on the hot path. It's only used where we parse the command line arguments/environment variables: https://github.com/Kong/kubernetes-ingress-controller/blob/18822a26b33f4452783f5af3819f2d0dd4795886/internal/manager/config.go#L183. Which make the performance gain from this patch negligible.

At the cost of this change, we've exposed a `var` that has a potential flaw of being writable by code within this package. Nothing critical as long as it's not misused but I wanted to call out that it can be used incorrectly which might cause surprises down the line.

Reply:
On the other hand, having something with the idiom Must as pkg level variables ensure that in case of a mistake (in this example in regex) it will panic as fast as the package is imported, so basically during the start-up of the program and mistake will be caught. But I know that it should/is covered with tests anyway, so it's not a crucial benefit